### PR TITLE
support for a single BS with multiple synced USRPs

### PR DESCRIPTION
**Pull request by tingjunchen**
In this PR:
- Added support to use multiple synced USRPs as the BS
- Fixed timing issues for USRPs during the start-up period when trigger by external PPS pulses
- Added support to enable using multiple USRPs in a single thread
- Tested in a 4x2 MIMO setting using one BS (two x310s with 4 channels in total, synced using octo-g) and one client (one N310 with 2 channels) for up to 25MHz real-time bandwidth

TODO:
- Add Hadamard matrix and beamsweep functionalities on the host

### CODE REVIEW

File: CC/Sounder/receiver.cc [hunk 1]
```diff
@@ -185,26 +185,44 @@ void Receiver::loopRecv(int tid, int core_id, SampleBuffer* rx_buffer)
...
-        baseRadioSet_->radioTx(0, 0, beaconbuff.data(), 2, txTimeBs);
-        long long bsInitRxOffset = txTimeBs - rxTimeBs;
-        for (int it = 0; it < std::floor(bsInitRxOffset / config_->sampsPerSymbol); it++) {
-            baseRadioSet_->radioRx(0, 0, samp_buffer.data(), config_->sampsPerSymbol, rxTimeBs);
+        // For multi-USRP BS perform dummy radioRx to avoid initial late packets
+        int bs_sync_ret = -1;
+        std::cout << "Sync BS host and FPGA timestamp for thread " << tid << "..." << std::endl;
+        for (int it = radio_start; it < radio_end; it++) {
+            // Find cell this USRP belongs to..,
+            for (int i = 0; i <= static_cast<int>(config_->nCells); i++) {
```
Comment (rdoosty) [inline CC/Sounder/receiver.cc:195]: Can we use `size_t` type for `i`? and `it`?

Comment (tingjunchen) [reply CC/Sounder/receiver.cc:195]: I think it should work, but I simply just copied this part from
https://github.com/renew-wireless/RENEWLab/blob/develop/CC/Sounder/receiver.cc#L212-L218

Should we use `size_t` for in those lines as well?

Comment (rdoosty) [reply CC/Sounder/receiver.cc:195]: Yes please. I understand there are instances in the code that is not doing this but we are gradually moving toward the Agora code style, with one of the points being to not use `int` in loops etc, unless it is absolutely needed.

Comment (tingjunchen) [reply CC/Sounder/receiver.cc]: done! tried to replaced `int` with `size_t` and avoid static cast in a few places

**Review by rdoosty — APPROVED** [review]

**Pull request merged**